### DISCUSSION TITLE
Problem running bdfr with subprocess.Popen

### DISCUSSION THREAD

**Discussion by Firestorm2k7** [Serene-Arc/bulk-downloader-for-reddit#569] [Q&A] [answered]
Hello everyone,
I'm very new to python and am having an issue running BDFR from my script as a subprocess (or in any orher way).

EDIT: I am running on Windows. Python Version is 3.10.0 and BDFR version is v2.5.1
EDIT2: Problem solved-ish. read first answer before investing too much time. Still curious what I originally did wrong.

The line I'm using to run it is this:

```
subprocess.Popen(
        [sys.executable,
         "C:/Users/<User>/AppData/Roaming/Python/Python310/site-packages/bdfr/__main__.py",
         "download", "C:/downloads", "-s playstupidgames", "-L 10"])
```

The Error I'm getting is this:
[2021-11-30 11:33:04,948 - bdfr.connector - ERROR] - Source  playstupidgames does not exist or cannot be found

If I execute
```
python "C:/Users/<User>/AppData/Roaming/Python/Python310/site-packages/bdfr/__main__.py" download "C:/downloads" -s playstupidgames -L 10
```
from my command line it works just fine

I'm absolutely not set on this specific solution, so I'm definitely open to change it.
Also I'll want to have it execute with modular arguments, but that's a problem for later.
A little help would br really appreciated, have a nice day ^^

**Comment by Firestorm2k7** [comment, marked as answer]
I am utterly confused, but I've accidentally got it to work Oo
I've used a bastardized version of some code to implement variable arguments while waiting:

```
# populate list of arguments
args = ["C:/Users/<User>/AppData/Roaming/Python/Python310/site-packages/bdfr/__main__.py"]

args.extend("download".split())
args.extend("C:/downloads".split())
args.extend("-s playstupidgames".split())
args.extend("-L 10".split())

# run script
p = subprocess.Popen([sys.executable] + args)
```

(copied and changed from https://stackoverflow.com/questions/20140137/passing-variables-to-subprocess-popen )

I expected to fiddle around with that for a bit and expect the same "does not exist or cannot be found" error as a positive outcome, but this version works and downloads the files. If someone wants to tell my what was wrong with the first veriant, I'd be happy to know.
Otherwise I guess this issue is done.
Leaving it open to maybe find out what I originally did wrong.
Thanks for providing this amazing piece of work to the world!

**Reply by Serene-Arc** [reply]
The issue was probably the fact that, originally, you had `'-s playstupidgames'` as one string. Proper syntax for that command is to have each contiguous option as a separate string so `'-s', 'playstupidgames'`. Later on you split based on the space character so that's probably why it ends up working since it does the same thing. So yeah if there's a space and it's supposed to an option and response, it needs to be separated.

**Reply by Firestorm2k7** [reply]
Oh, good to know, thanks for the clarification! ^^
